packages/health/android/src/main/kotlin/cachet/plugins/health/HealthPlugin.kt

```diff
@@ -1691,15 +1691,37 @@ class HealthPlugin(private var channel: MethodChannel? = null) :
         val healthConnectData = mutableListOf<Map<String, Any?>>()
         scope.launch {
             MapToHCType[dataType]?.let { classType ->
-                val request = ReadRecordsRequest(
+                val records = mutableListOf<Record>()
+
+                // Set up the initial request to read health records with specified parameters
+                var request = ReadRecordsRequest(
                     recordType = classType,
+                    // Define the maximum amount of data that HealthConnect can return in a single request
                     timeRangeFilter = TimeRangeFilter.between(startTime, endTime),
                 )
-                val response = healthConnectClient.readRecords(request)
+                
+                var response = healthConnectClient.readRecords(request)
+                var pageToken = response.pageToken
+                
+                // Add the records from the initial response to the records list
+                records.addAll(response.records)
+
+                // Continue making requests and fetching records while there is a page token
+                while (!pageToken.isNullOrEmpty()) {
```

Comment:
I mainly made this small change, since I was receiving an empty string instead of a null string, for whatever reason.

Reply:
Oops! I applied that change to our fork but totally forgot to do the same thing here. It's correct that the line has to be changed as you did it. Providing more context, the page token is actually [documented](https://developer.android.com/reference/androidx/health/connect/client/response/ReadRecordsResponse#summary) to be null if there are no pages left. However, this behavior differs between Android 13 and Android 14. The phone I was testing with was running Android 14, so it returned null. Nevertheless, we encountered the same issue of an infinite loop with older devices.

Reply:
Great, I will merge it into my branch then and include it in the upcoming release. Thanks a lot!